core/app/models/spree/shipment.rb

```diff
@@ -241,6 +240,20 @@ def selected_shipping_rate_id=(id)
       self.save!
     end
 
+    def process_inventory_unit_list(inventory_list = [])
+      conn = ActiveRecord::Base.connection
+      _time = Time.zone.now.to_s(:db)
+      insert_clause  = %Q{ INSERT INTO #{Spree::InventoryUnit.quoted_table_name} (#{conn.quote_column_name("state")}, #{conn.quote_column_name("variant_id")}, #{conn.quote_column_name("order_id")}, #{conn.quote_column_name("shipment_id")}, #{conn.quote_column_name("created_at")}, #{conn.quote_column_name("updated_at")}, #{conn.quote_column_name("line_item_id")}) VALUES }
+      inventory_list.group_by(&:state).each do |state, inventory_list_items|
+        inventory_list_items.group_by(&:variant_id).each do |variant_id, items|
+          values = items.map do |item|
+            %Q{ ('#{ state }', #{ variant_id }, #{ self.order_id }, #{ self.id }, '#{ _time }', '#{ _time }', #{ item.line_item_id }) }
+          end.join(", ")
+          conn.execute(insert_clause + values)
+        end
+      end
```

Comment:
@nimish-mehta We should reload `inventory_units` at the end of this method to ensure that we get inserted units.